enhancements/storage/csi-ebs-operator.md

```diff
@@ -0,0 +1,142 @@
+---
+title: csi-ebs-operator
+authors:
+  - "@fbertina"
+reviewers:
+  - "@jsafrane”
+  - "@hekumar"
+  - "@chuffman"
+approvers:
+  - "@..."
+creation-date: 2020-03-09
+last-updated: 2020-03-09
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# CSI Operator for AWS EVS
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+In the past storage drivers have been provided either in-tree or via external provisioners. With the move to advocating CSI as the way to deliver storage drivers, Red Hat needs to move the in-tree drivers (and others that our customers need) to use CSI. With the focus on operators in Red Hat OCP 4.x these CSI drivers should be installed in the form of an Operator.
+
+## Motivation
+
+* In-tree drivers will be removed from Kubernetes, we need to continue to make the drivers available and CSI is the way to do this.
+* AWS is a key cloud provider for OpenShift and use of EBS has been supported in both OpenShift 3.x and 4.x, we need to move this driver to be CSI, provided by an operator.
+* CSI drivers (incl. AWS EBS) will enable new storage features, not possible with in-tree volume plugins, like snapshots and volume cloning.
+
+### Goals
+
+* Create an operator to install the AWS EBS CSI driver.
+* Publish the operator on OperatorHub.
+* Package and ship downstream image of AWS EBS CSI driver (upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
+
+### Non-Goals
+
+* Driver creation (upstream: https://github.com/kubernetes-sigs/aws-ebs-csi-driver)
+
+## Proposal
+
+1. OCP ships a new operator called aws-ebs-csi-driver-operator. This operator:
+   * Installs the AWS EBS CSI driver.
+	 * It creates a new Deployment that runs the controller pieces of the driver (provisioner, attacher, resizer, etc.).
+	 * It creates a new DaemonSet that runs the CSI driver on the worker nodes.
```

Comment:
Assuming that namespace for the driver is created by user / OLM. Who & when creates ServiceAccount, Roles, RoleBindings, ClusterRoleBindings and so on for the *driver*?

Reply:
Currently there is no way for OLM to create the target namespace and it must be created by the user before installing the operator. The catalog-operator should then copy  SA,RBAC rules etc in target namespace.

Reply:
I know that OLM creates SA & RBAC rules for the *operator*. Can it create the same for the *driver*? Because the RBAC will be different.

Reply:
afaict - it can't create RBAC rules for the driver, esp because driver is installed in a different namespace than the operator and OLM merely applies RBAC rules specified in CSV to the namespace in which operator is installed.